pyacq/core/manager.py

```diff
@@ -9,46 +9,102 @@ class Manager(RPCServer):
+        
+        # shared socket for all RPC client connections
         self._rpc_socket = RPCClientSocket()
 
     def connect_host(self, name, addr):
-        if name not in self._hosts:
-            hp = HostProxy(self, name, addr)
-            self._hosts[name] = hp
-        return self._hosts[name]
+        if name not in self.hosts:
+            hp = _Host(name, addr)
+            self.hosts[name] = hp
 
     def disconnect_host(self, name):
-        pass
+        self.hosts[name].close()
+
+    def list_hosts(self):
+        return list(self.hosts.keys())
+    
+    def create_nodegroup(self, host, name):
+        if name in self.nodegroups:
+            raise KeyError("Nodegroup named %s already exists" % name)
+        host = self.hosts[host]
+        addr = 'tcp://%s:*' % (host.rpc_hostname)
+        _, addr = host.client.new_nodegroup(name, addr)
+        ng = _NodeGroup(host, name, addr)
+        host.add_nodegroup(ng)
+        self.nodegroups[name] = ng
+
+    def list_nodegroups(self, host=None):
+        if host is None:
+            return list(self.nodegroups.keys())
+        else:
+            return self.hosts[host].list_nodegroups()
+
+    def create_node(self, nodegroup, name, classname, **kwargs):
+        if name in self.nodes:
+            raise KeyError("Node named %s already exists" % name)
+        ng = self.nodegroups[nodegroup]
+        ng.client.create_node(name, classname, **kwargs)
+        node = _Node(ng, name, classname)
+        self.nodes[name] = node
+        ng.add_node(name, node)
 
-    def add_nodegroup(self, host, name):
-        addr = ...
-        self._hosts[host].client.new_nodegroup(name, addr)
-        ng = NodeGroupProxy(self, name, addr)
-        self._nodegroups[name] = ng
+    def list_nodes(self, nodegroup=None):
+        if nodegroup is None:
+            return list(self.nodes.keys())
+        else:
+            return self.nodegroups[nodegroup].list_nodes()
 
-    def select_free_address(self, host, prot='tcp'):
-        pass
+    def control_node(self, name, method, **kwargs):
+        ng = self.nodes[name].nodegroup
+        return ng.client.control_node(name, method, **kwargs)
+    
+    def delete_node(self, name):
+        ng = self.nodes[name].nodegroup
+        ng.client.delete_node(name)
+        del self.nodes[name]
+        ng.delete_node(name)
 
         
-class HostProxy(object):
-    def __init__(self, mgr, name, addr):
-        self.mgr = mgr
+class _Host(object):
+    def __init__(self, name, addr):
         self.rpc_address = addr
         self.rpc_name = name
         self.client = RPCClient(name, addr)
+        self.nodegroups = {}
+        self.rpc_hostname = addr.partition('//')[2].rpartition(':')[0]
 
-    def add_nodegroup(self, name):
-        return self.mgr.add_nodegroup(self.rpc_name, name)
+    def add_nodegroup(self, ng):
+        self.nodegroups[ng.rpc_name] = ng
+    
+    def list_nodegroups(self):
+        return list(self.nodegroups.keys())
 
```

Comment:
By design this _Host class is too much, no ? This do redundently what pycaq.core.host.Host.
Could we have a automagic way ?
But we can leave it for the moement.